python/lsst/pipe/tasks/configurableActions/_configurableActionStructField.py

```diff
@@ -0,0 +1,330 @@
+        else:
+            raise ValueError("Can only update a ConfigurableActionStruct with an instance of such, or a "
+                             "mapping")
+        for name, action in value.items():
+            setattr(instance, name, action)
+
+    def __get__(self, instance, objtype=None) -> None:
+        # This descriptor does not support fetching any value
+        return None
+
+
+class ConfigurableActionStructRemover:
+    """This descriptor exists to abstract the logic of removing an interable
+    of action names from a ConfigurableActionStruct at one time using
+    attribute assignment. This is useful in the context of setting
+    configuration through pipelines or on the command line.
+
+    Raises
+    ------
+    AttributeError
+        Raised if an attribute specified for removal does not exist in the
+        ConfigurableActionStruct
+    """
+    def __set__(self, instance: ConfigurableActionStruct,
+                value: Union[str, Iterable[str]]) -> None:
+        # strings are iterable, but not in the way that is intended. If a
+        # single name is specified, turn it into a tuple before attempting
+        # to remove the attribute
+        if isinstance(value, str):
+            value = (value, )
+        for name in value:
+            delattr(instance, name)
+
+    def __get__(self, instance, objtype=None) -> None:
+        # This descriptor does not support fetching any value
+        return None
+
+
+class ConfigurableActionStruct:
+    """A ConfigurableActionStruct is the storage backend class that supports
+    the ConfigurableActionStructField. This class should not be created
+    directly.
+
+    This class allows managing a collection of `ConfigurableActions` with a
+    struct like interface, that is to say in an attribute like notation.
+
+    Attributes can be dynamically added or removed as such:
+
+    ConfigurableActionStructInstance.variable1 = a_configurable_action
+    del ConfigurableActionStructInstance.variable1
+
+    Each action is then available to be individually configured as a normal
+    `lsst.pex.config.Config` object.
+
+    ConfigurableActionStruct supports two special convenance attributes.
+
+    The first is `update`. You may assign a dict of `ConfigurableActions` or
+    a `ConfigurableActionStruct` to this attribute which will update the
+    `ConfigurableActionStruct` on which the attribute is invoked such that it
+    will be updated to contain the entries specified by the structure on the
+    right hand side of the equals sign.
+
+    The second convenience attribute is named remove. You may assign an
+    iterable of strings which correspond to attribute names on the
+    `ConfigurableActionStruct`. All of the corresponding attributes will then
+    be removed. If any attribute does not exist, an `AttributeError` will be
+    raised. Any attributes in the Iterable prior to the name which raises will
+    have been removed from the `ConfigurableActionStruct`
```

Comment:
Why special decorators instead of just methods?

I think e.g.
```
config.my_action.update(dictionary)
```
reads a lot more naturally than
```
config.my_action.update = dictionary
```
Is this just for command-line/pipeline overrides again?  If so, they're giving me some reservations about whether catering to their needs is a good idea.